### PR TITLE
feat: add prettier

### PR DESCRIPTION
**Pull request by henryiii**
Should be implemented in Cookie first.

### CODE REVIEW

**Comment by eduardo-rodrigues** [comment]
How does this interfere, if at all, with for example Black? This point may be worth a sentence.

**Comment by henryiii** [comment]
Pretty sure Python is not one of the languages it supports. But that's partially why it has a mention of using `types_or` to ensure it's not trying to change files you'd rather not be auto formatted. I should include json in the list. I'm pretty sure we don't already recommend an auto formatter for any of the supported file types.

Screen Shot 2022-05-05 at 4 46 31 PM

File: pages/developers/style.md [hunk 1]
```diff
@@ -532,6 +532,26 @@ If you have shell scripts, you can protect against common mistakes using [shellc
...
+```yaml
+  - repo: https://github.com/pre-commit/mirrors-prettier
+    rev: "v2.6.2"
+    hooks:
+      - id: prettier
+        types_or: [yaml, markdown, html, js]
+```
+
+Since this formats a variety of very common file types (like `html`, `md`,
+`yaml`, `js`, and more), you will usually want to provide a `types_or` setting
```
Comment (henryiii) [inline pages/developers/style.md:549]:
```suggestion
`yaml`, `js`, `json`, and more), you will usually want to provide a `types_or` setting
```

File: pages/developers/style.md [hunk 2]
```diff
@@ -532,6 +532,26 @@ If you have shell scripts, you can protect against common mistakes using [shellc
...
+
+The [prettier](https://prettier.io) tool can format a large number of different
+file types. An example of usage:
+
+```yaml
+  - repo: https://github.com/pre-commit/mirrors-prettier
+    rev: "v2.6.2"
+    hooks:
+      - id: prettier
+        types_or: [yaml, markdown, html, js]
```
Comment (henryiii) [inline pages/developers/style.md:545]:
```suggestion
        types_or: [yaml, markdown, html, js, json]
```

Comment (eduardo-rodrigues) [reply pages/developers/style.md:545]: Got it. Thanks.

**Comment by henryiii** [comment]
Updates: I've run prettier on the website itself, and worked out a few things:

* Prettier can format YAML inside codeblocks in markdown! That's great, so I added blacken-docs which does the same thing for Python (Black) inside markdown.
    * This actually found a bug, we had invalid syntax in one example!
* I'm having prettier reflow everything to ~80 chars. We could also have it reflow to no line wrap, I care more about consistency than which setting.
* Pretty _mostly_ is okay with Jekyll, but it does break `{: .no_toc}` or similar right after a heading by adding a blank line. I've avoided this most places by adding the title directly to the page - you can override the title with `custom_title: <>`. If you add `custom_title` but leave it blank, there will be no title at the top.
* I've replaced the ugly capture workarounds with the correct `markdown="1"` for expandable. Later I'll style the expandable to look more like the current TOC styling.
* There was some slightly broken or non standard YAML that I've cleaned up with prettier's help.

**Review by eduardo-rodrigues — APPROVED** [review]
Difficult to navigate the many tiny changes but all seems fine.

**Pull request merged**